de.fu_berlin.inf.dpp.intellij/src/de/fu_berlin/inf/dpp/intellij/filesystem/VirtualFileConverter.java

```diff
@@ -0,0 +1,169 @@
+    }
+
+    private VirtualFileConverter() {
+        //NOP
+    }
+
+    /**
+     * Returns an <code>SPath</code> representing the given file.
+     *
+     * @param virtualFile file to get the <code>SPath</code> for
+     * @return an <code>SPath</code> representing the given file or
+     * <code>null</code> if given file does not exist, no module could be found
+     * for the file or the found module can not be shared through saros, or the
+     * relative path between the module root and the file could not be
+     * constructed
+     */
+    @Nullable
+    public static SPath convertToSPath(
+        @NotNull
+            VirtualFile virtualFile) {
+
+        IResource resource = convertToResource(virtualFile);
+
+        return resource == null ? null : new SPath(resource);
+    }
+
+    /**
+     * Returns an <code>IResource</code> representing the given
+     * <code>VirtualFile</code>.
+     *
+     * @param virtualFile file to get the <code>IResource</code> for
+     * @return an <code>IResource</code> representing the given file or
+     * <code>null</code> if given file does not exist, no module could be found
+     * for the file or the found module can not be shared through saros, or the
+     * relative path between the module root and the file could not be
+     * constructed
+     */
+    @Nullable
+    public static IResource convertToResource(
+        @NotNull
+            VirtualFile virtualFile) {
+
+        Module module = ModuleUtil.findModuleForFile(virtualFile, project);
+
+        if (module == null) {
+            log.warn("Could not convert VirtualFile " + virtualFile
+                + " as no module could be found for the file.");
+
+            return null;
+        }
+
+        IntelliJProjectImplV2 sarosProject;
+
+        try {
+            sarosProject = new IntelliJProjectImplV2(module);
+
+        } catch (IllegalArgumentException | IllegalStateException e) {
+            log.debug("Could not convert VirtualFile " + virtualFile
+                + " as the creation of an IProject object for its module "
+                + module + " failed.", e);
+
+            return null;
+        }
+
+        return sarosProject.getResource(virtualFile);
+    }
+
+    /**
+     * Returns an <code>IResource</code> representing the given
+     * <code>VirtualFile</code>.
+     *
+     * @param virtualFile file to get the <code>IResource</code> for
+     * @param project     module the file belongs to
+     * @return an <code>IResource</code> for the given file or
+     * <code>null</code> if the given file does not exist, does not belong to
+     * the passed module, or the relative path path between the module root and
+     * the file could not be constructed
+     */
+    @Nullable
+    public static IResource convertToResource(
+        @NotNull
+            VirtualFile virtualFile,
+        @NotNull
+            IProject project) {
+
+        IntelliJProjectImplV2 module = (IntelliJProjectImplV2) project
```

Comment:
Either module or this "fancy" sarosXYZ name, but do not mix it up, see all other methods.

Reply:
Done. Renamed all to wrappedModule.